File: pyg_parent/pyg_interface/src/main/java/com/itheima/pyg/service/brand/BrandService.java
package com.itheima.pyg.service.brand;

import com.itheima.pyg.entity.PageResult;
import com.itheima.pyg.entity.Result;
import com.itheima.pyg.pojo.good.Brand;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询全部品牌
     * @return
     */
    List<Brand> findAllBrands();

    /**
     * 分页查询品牌
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findBrandListByPage(Integer pageNum,Integer pageSize);

    /**
     * 根据条件分页查询
     * @param pageNum
     * @param pageSize
     * @param brand
     * @return
     */
    PageResult  searchBrands(Integer pageNum,Integer pageSize,Brand brand);

    /**
     * 添加品牌
     * @param brand
     * @return
     */
    Result  addBrand(Brand brand);

    /**
     * 根据id查询品牌
     * @param id
     * @return
     */
    Brand findOne(long id);

    /**
     * 更新品牌
     * @param brand
     */
    void updateBrand(Brand brand);

    /**
     * 删除品牌
     * @param ids
     */
    void deleteBrandsByIds(long[] ids);

    /**
     * 查询品牌列表项,用作模板关联品牌新增时下拉框数据显示
     * @return
     */
    List<Map<String,String>> selectOptionList();
}

File: pyg_parent/pyg_web_manager/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAllBrands.action");
	}
	
	this.findPage = function(page,rows){
		return $http.get("../brand/findBrandListByPage.action?pageNum="+page+"&pageSize="+rows);
	}
	
	this.add = function(entity){
		return $http.post("../brand/addBrand.action",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/updateBrand.action",entity);
	}
	
	this.findOne=function(id){
		return $http.get("../brand/findOne.action?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/deleteBrandsByIds.action?ids="+ids);
	}
	
	this.search = function(page,rows,searchEntity){
		return $http.post("../brand/searchBrands.action?pageNum="+page+"&pageSize="+rows,searchEntity);
	}
	
	this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList.action");
	}
});